### PR TITLE
network: delete IP addrs on bridge model to prevent ARP conflict

### PR DESCRIPTION
**Pull request by zhabinecho**
Fixes: #1673

Signed-off-by: Zha Bin <zhabin@linux.alibaba.com>

### CODE REVIEW

File: virtcontainers/network.go [hunk 1]
```diff
@@ -1001,7 +1011,9 @@ func unBridgeNetworkPair(endpoint Endpoint) error {
 		return fmt.Errorf("Could not detach veth %s: %s", netPair.VirtIface.Name, err)
 	}
 
-	return nil
+	// Restore the IPs that were cleared
+	err = setIPs(link, netPair.VirtIface.Addrs)
+	return err
```
Comment (jodh-intel) [inline virtcontainers/network.go]: Nit: you could simplify this to:

```go
return setIPs(link, netPair.VirtIface.Addrs)
```

**Comment by jodh-intel** [comment]
Restarted the containerd CI which failed with what looks like a transitory error:

```
  failed to pull image: rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/library/busybox:1.28": failed to unpack image on snapshotter overlayfs: failed to extract layer sha256:432b65032b9466b4dadcc5c7b11701e71d21c18400aae946b101ad16be62333a: mount callback failed on /var/lib/containerd/tmpmounts/containerd-mount743870557: link /var/lib/containerd/tmpmounts/containerd-mount743870557/bin/[ /var/lib/containerd/tmpmounts/containerd-mount743870557/bin/runsv: no such file or directory: unknown
    Expected error:
        <*status.statusError | 0xc000528510>: {
            Code: 2,
            Message: "failed to pull and unpack image \"docker.io/library/busybox:1.28\": failed to unpack image on snapshotter overlayfs: failed to extract layer sha256:432b65032b9466b4dadcc5c7b11701e71d21c18400aae946b101ad16be62333a: mount callback failed on /var/lib/containerd/tmpmounts/containerd-mount743870557: link /var/lib/containerd/tmpmounts/containerd-mount743870557/bin/[ /var/lib/containerd/tmpmounts/containerd-mount743870557/bin/runsv: no such file or directory: unknown",
            Details: nil,
        }
        rpc error: code = Unknown desc = failed to pull and unpack image "docker.io/library/busybox:1.28": failed to unpack image on snapshotter overlayfs: failed to extract layer sha256:432b65032b9466b4dadcc5c7b11701e71d21c18400aae946b101ad16be62333a: mount callback failed on /var/lib/containerd/tmpmounts/containerd-mount743870557: link /var/lib/containerd/tmpmounts/containerd-mount743870557/bin/[ /var/lib/containerd/tmpmounts/containerd-mount743870557/bin/runsv: no such file or directory: unknown
    not to have occurred
```

File: virtcontainers/network.go (hunk 1 above)
Comment (zhabinecho) [reply virtcontainers/network.go]: Fixed it :-)

Comment (jodh-intel) [reply virtcontainers/network.go]: Thanks @zhabinecho!

**Comment by amshinde** [comment]
this is already handled for the macvtap mode, somehow we missed it in the bridge mode.
@zhabinecho this lgtm,  but I would like to understand if you are using the bridge mode at all, we are planning to get rid of the bridge mode. There is a PR open for it, I plan to push that soon.
Before that, I would like to hear from you, if you are fine with this, the we recommend switcing to the `tcfilter` mode to avoid any issues.

**Pull request merged**